src/dune_rules/odoc.ml

```diff
@@ -203,7 +211,29 @@ let compile_module sctx ~obj_dir (m : Module.t) ~includes:(file_deps, iflags)
       ] );
   (m, odoc_file)
 
-let compile_mld sctx (m : Mld.t) ~includes ~doc_dir ~pkg =
+let compile_odoc_to_odocl sctx ~obj_dir (m : Module.t)
```

Comment:
This appears to be identical (except in name) to the above function

Reply:
Ah, I see, you've got the odocl file there. In any case, the function 'link_odoc_file' looks much better than these two functions - I think you should get rid of these, they don't look right at all.

Reply:
If I understand correctly, we need such functions. They appear to be the same, but the targets and dependencies differ. FYI, in such a case, `(m : Module.t)` this is for determining the source of `odoc` file.